qiskit/pulse/library/symbolic_pulses.py

```diff
@@ -83,20 +83,24 @@ def _lifted_gaussian(
 
 
 @functools.lru_cache(maxsize=None)
-def _is_amplitude_valid(symbolic_pulse: "SymbolicPulse") -> bool:
+def _is_amplitude_valid(envelope_lam: Callable, time: Tuple[float, ...], *fargs: float) -> bool:
     """A helper function to validate maximum amplitude limit.
 
     Result is cached for better performance.
 
     Args:
-        symbolic_pulse: A pulse to validate.
+        envelope_lam: The SymbolicPulse's lambdified envelope_lam expression.
+        envelope: The SymbolicPulse's envelope expressions.
+        parameters: The SymbolicPulse's parameters.items() (assumed to be binded) converted
+            to tuple (for hashability).
 
     Returns:
         Return True if no sample point exceeds 1.0 in absolute value.
     """
     try:
+        time = np.asarray(time, dtype=float)
         # Instantiation of Waveform does automatic amplitude validation.
-        symbolic_pulse.get_waveform()
+        Waveform(samples=envelope_lam(time, *fargs))
```

Comment:
Just checking -- is it still best to use `Waveform` for this check now that we are calculating the array directly? We could just check if all samples are below 1 (or 1+1e-7 like `Waveform` does).

Reply:
Technically, nothing has changed compared to what we did before. It could be argued that it's best to leave the test logic in a single place, as it is not trivial. I am fine with both options.

Reply:
I agree with Will. We can directly test amplitude limit, which should be slightly faster.